libs/ledger/ledger/injections.py

```diff
@@ -46,8 +69,63 @@ def mass_2_source(self):
         return self.mass_2 / (1 + self.redshift)
 
     @property
-    def chirp_mass(self):
-        return chirp_mass(self.mass_1, self.mass_2)
+    def luminosity_distance(self, cosmology=DEFAULT_COSMOLOGY):
+        return cosmology.luminosity_distance(self.redshift).value
+
+
+@dataclass
+class PycbcParameterSet(ExtrinsicParameterSet, IntrinsicParameterSet):
+    def convert_to_lal_params(self, reference_frequency: float):
+        self.inclination = np.zeros(len(self))
+        self.spin1x = np.zeros(len(self))
+        self.spin1y = np.zeros(len(self))
+        self.spin1z = np.zeros(len(self))
+        self.spin2x = np.zeros(len(self))
+        self.spin2y = np.zeros(len(self))
+        self.spin2z = np.zeros(len(self))
+
+        for i in range(len(self)):
+            (
+                self.inclination[i],
+                self.spin1x[i],
+                self.spin1y[i],
+                self.spin1z[i],
+                self.spin2x[i],
+                self.spin2y[i],
+                self.spin2z[i],
+            ) = bilby_to_lalsimulation_spins(
```

Comment:
I think bilby supports a vectorized form of this: https://git.ligo.org/lscsoft/bilby/-/blob/c6bcb81649b7ebf97ae6e1fd689e8712fe028eb0/bilby/gw/conversion.py#L163

Reply:
Yeah, weirdly it doesn't work because of [this line](https://git.ligo.org/lscsoft/bilby/-/blob/c6bcb81649b7ebf97ae6e1fd689e8712fe028eb0/bilby/gw/conversion.py#L145) at the start of the function. The comparison doesn't work for arrays.

Reply:
ah, thats a bit annoying - alright maybe let's open up an to do our own conversion at some point.

Reply:
The calculation was a little involved, which is why I didn't try it here. But agreed, it silly to have `bilby` as a dep just to get this function